lsp/src/saros/lsp/SarosLauncher.java

```diff
@@ -1,61 +1,216 @@
 package saros.lsp;
 
+import java.io.File;
 import java.io.IOException;
-import java.net.Socket;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.channels.AsynchronousServerSocketChannel;
+import java.nio.channels.AsynchronousSocketChannel;
+import java.nio.channels.Channels;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.function.Function;
+import org.apache.log4j.Level;
 import org.apache.log4j.Logger;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.apache.logging.log4j.core.lookup.MainMapLookup;
+import org.apache.logging.log4j.status.StatusLogger;
 import org.eclipse.lsp4j.jsonrpc.Launcher;
+import org.eclipse.lsp4j.jsonrpc.MessageConsumer;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
 import saros.lsp.extensions.client.ISarosLanguageClient;
+import saros.lsp.extensions.server.ISarosLanguageServer;
+import saros.lsp.filesystem.IWorkspacePath;
+import saros.lsp.filesystem.WorkspacePath;
 import saros.lsp.log.LanguageClientAppender;
 
-/** Entry point for the Saros LSP server. */
-public class SarosLauncher {
+/** Entry point for the Saros language server. */
+public class SarosLauncher implements Callable<Integer> {
 
-  private static final Logger log = Logger.getLogger(SarosLauncher.class);
+  private static final Logger LOG = Logger.getLogger(SarosLauncher.class);
+  private static final String LOG4J2_CONFIG_FILENAME = "saros_log4j2.xml";
+
+  @Option(
+      names = {"-p", "--port"},
+      description = "The port to listen on")
+  int port;
+
+  @Option(
+      names = {"-l", "--log"},
+      description = "The log level")
+  String logLevel;
+
+  @Override
+  public Integer call() throws Exception {
+    setupLoggers();
+    LOG.info("listening on port " + port);
+
+    final AsynchronousServerSocketChannel serverSocket =
```

Comment:
Where is this serverSocket stored ? Otherwise the socket will be closed if it goes out of scope and the GC will trash it afterwards.